src/handlers/validate_config.rs

```diff
@@ -0,0 +1,117 @@
+//! For pull requests that have changed the triagebot.toml, validate that the
+//! changes are a valid configuration file.
+//! It won't validate anything unless the PR is open and has changed.
+//! For every event, it will fetch the diff of the pull request which is one
+//! round-trip of HTTP request to the GitHub API.
+
+use crate::{
+    config::ValidateConfig,
+    github::IssuesAction,
+    handlers::{Context, IssuesEvent},
+};
+use tracing as log;
+
+/// If the issue is a pull request and if among the changed files, triagebot.toml
+/// is changed, then validate the triagebot.toml config file with the syntax
+/// parser from the config module.
+pub(super) async fn parse_input(
+    ctx: &Context,
+    event: &IssuesEvent,
+    _config: Option<&ValidateConfig>,
+) -> Result<Option<()>, String> {
+    if !event.issue.is_pr() {
+        log::debug!("Ignoring issue: {:?}", event.issue);
+        return Ok(None);
+    }
+
+    if !matches!(
+        event.action,
+        IssuesAction::Opened | IssuesAction::Synchronize | IssuesAction::ReadyForReview,
+    ) {
+        log::debug!("Ignoring issue action: {:?}", event.action);
+        return Ok(None);
+    }
+
+    let diff = match event.issue.diff(&ctx.github).await {
+        Ok(Some(diff)) => diff,
+        wildcard => {
+            log::error!("Failed to fetch diff: {:?}", wildcard);
+            return Ok(None);
+        }
+    };
+
+    if diff.contains(crate::config::CONFIG_FILE_NAME) {
```

Comment:
I have a few concerns about this. IIUC, this fetches the full diff on every PR on every push. That seems like it would be expensive, and potentially slow.

Would it be possible to use the `files()` method instead? I would imagine that is more efficient, and is also structured.

Similarly, this looks like it does a string `contains` check, which means that if any text anywhere in the patch contains the bytes `triagebot.toml`, it would trigger. If it looked at the structured file list, then it should be able to look to see if that file is specifically being modified.

I'm still not sure if hitting the `files` endpoint on every push is OK, but I suspect the traffic should be low enough to be fine.